docs/design.md

```diff
@@ -200,22 +211,9 @@ Transactions are executed in two phases:
    transaction table when there are conflicts and to make
    tie-breaking decisions on ordering between identical timestamps.
    Each node returns the timestamp used for the write (which is the
-   original candidate timestamp in the absence of conflicts); the client
-   selects the maximum from amongst all writes as the final commit
-   timestamp.
-
-   Each range maintains a small (i.e. latest 10s of read timestamps),
-   *in-memory* cache from key to the latest timestamp at which the
-   key(s) were read. This *latest-read-cache* is consulted on each
-   write. If the write’s candidate timestamp is earlier than the low
-   water mark on the cache itself (i.e. its last evicted timestamp)
-   or if the key being written has a read timestamp later than the
-   write’s candidate timestamp, this later timestamp value is
-   returned with the write. The cache’s entries are evicted oldest
-   timestamp first, updating low water mark as appropriate. If a new
-   range replica leader is elected, it sets the low water mark for
-   the cache to the current wall time + ε (ε = 99^th^ percentile
-   clock skew).
```

Comment:
I've split this section into two sections. One above describing the book keeping mechanism, and the section below describing the conflict and how it is resolved.